sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,233 @@
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      { student },
+      { createSecondDisbursement: true },
+    );
+    const endpoint = `/students/application/${application.id}/completed`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: 404,
+        message: "Application not found or not on Completed status.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should get application details when application is in 'Completed' status.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      { student },
+      { createSecondDisbursement: true },
+    );
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement, secondDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    const endpoint = `/students/application/${application.id}/completed`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        firstDisbursement: {
+          coeStatus: firstDisbursement.coeStatus,
+          disbursementScheduleStatus:
+            firstDisbursement.disbursementScheduleStatus,
+        },
+        secondDisbursement: {
+          coeStatus: secondDisbursement.coeStatus,
+          disbursementScheduleStatus:
+            secondDisbursement.disbursementScheduleStatus,
+        },
+        assessmentTriggerType: application.currentAssessment.triggerType,
+      });
+  });
+
+  it(`Should get application details with scholastic standing change type when application has a scholastic standing '${StudentScholasticStandingChangeType.StudentDidNotCompleteProgram} associated with.`, async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(appDataSource, {
+      student,
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create a scholastic standing and have it associated with the completed application.
+    const scholasticStanding = createFakeStudentScholasticStanding({
```

Comment:
`application` can be passed here itself right?

Reply:
Changed.